### PR TITLE
Update AOSP building guide for Android 10

### PR DESCRIPTION
**Pull request by ZVNexus**
## Description
Adds a bit about ADB/Fastboot and their udev rules

### CODE REVIEW

File: software/building-android/en.md [hunk 1]
```diff
@@ -121,6 +121,31 @@ export CCACHE_COMPRESS=1
 
 *After making changes to your `~/.bashrc` file, remember to run `source ~/.bashrc` for changes to take effect.*
 
+## ADB/Fastboot
+
+You want to install "Android Tools" from the third party software center.
+
+### ADB udev rules
+``` bash
+sudo curl --create-dirs -L -o /etc/udev/rules.d/51-android.rules -O -L https://raw.githubusercontent.com/M0Rf30/android-udev-rules/master/51-android.rules
```
Comment (JoshStrobl) [inline software/building-android/en.md]: I do not feel comfortable with having people curl an arbitrary file onto their system and would much prefer more verbose instructions detailing the creation of the directory, as well as providing the rules file contents in the article for them to add.

File: software/building-android/en.md [hunk 2]
```diff
@@ -121,6 +121,31 @@ export CCACHE_COMPRESS=1
...
+```
+
+Set permissions
```
Comment (JoshStrobl) [inline software/building-android/en.md]: Unnecessary linebreaks and use of code blocks, can be summarized into:

```
1. Set permissions appropriately by doing `sudo chmod 00644 /etc/udev/rules.d/51-android.rules` and `sudo chown root /etc/udev/rules.d/51-android.rules`
2. Reload the udev rules by doing `sudo usysconf run udev-rules -f`

**Comment by ZVNexus** [comment]
@JoshStrobl Would you rather want to have it pick up VIM, or create a standalone xxd package for AOSP to use?

Also Android 10 now requires the CGPT tool, its apart of the Chromium project. I just steal the binary from a Debian package of Chrome tools and throw it in path and it works fine with just that, but that isn't an acceptable solution. I'll submit a package for CGPT later. "GPT manipulation tool with support for Chromium OS extensions" btw.

**Comment by ZVNexus** [comment]
Here's the package request for CGPT
https://dev.getsol.us/T8516

**Comment by ZVNexus** [comment]
vboot-utils is not currently in the repository, but I would _strongly_ recommend it as the current documentation is broken.

https://dev.getsol.us/T8516

**Comment by joebonrichie** [comment]
Manually added to new docusarus site, if there is anything else missing please create a new PR.

**Pull request closed without merging**